nntrainer/layers/layer_context.cpp

```diff
@@ -65,7 +65,7 @@ void InitLayerContext::setOutputDimensions(
   std::vector<VarGradSpecV2> specs;
   specs.reserve(out_dim.size());
 
-  for (unsigned i = 0u, sz = out_dim.size(); i < sz; ++i) {
+  for (unsigned i = 0u; i < out_dim.size(); ++i) {
```

Comment:
is calling std::vector<TensorDim>.size() repeatedly ok? (performance-wise)

Reply:
According to the cpp reference the complexity of size function is constant. So I guess it will be okay.
Please refer (https://en.cppreference.com/w/cpp/container/vector/size)

Reply:
Since the length can change inside the for loop, I think it's better to assign the length to a new variable for safety reasons.
In addition, i wonder in the code below, the length is changed through push_back, but wouldn't the result be different if sz is not defined?

```.cpp
  for (unsigned i = 0u, sz = out_specs.size(); i < sz; ++i) {
    auto &spec = out_specs.at(i);
    ....    
    ....
    output_specs.push_back(std::move(spec)); <--- change out_specs size 
  }
```

Reply:
The out_dim is const value so I guess length will not change inside the for loop but I will assign the length to a new variable any way.